qiskit/algorithms/time_evolvers/evolution_problem.py

```diff
@@ -0,0 +1,96 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Evolution problem class."""
+from __future__ import annotations
+from typing import Dict
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.opflow import PauliSumOp
+from ..list_or_dict import ListOrDict
+from ...quantum_info.operators.base_operator import BaseOperator
+
+
+class EvolutionProblem:
+    """Evolution problem class.
+
+    This class is the input to time evolution algorithms and must contain information on the total
+    evolution time, a quantum state to be evolved and under which Hamiltonian the state is evolved.
+    """
+
+    def __init__(
+        self,
+        hamiltonian: BaseOperator | PauliSumOp,
+        time: float,
+        initial_state: QuantumCircuit | None = None,
```

Comment:
We could also support `Statevector` here and convert it into a circuit by using `QuantumCircuit.prepare_state` if we want 🙂

Reply:
`prepare_state` is a method on the quantum circuit, so you can just it as 
```
circuit = QuantumCircuit(initial_state.num_qubits)
circuit.prepare_state(initial_state.data)
```
🙂 (side note: All circuit methods return an instruction set at the moment, that's used for classical conditions)